### PR TITLE
Add missing tests for attributesMap

### LINKED ISSUE #1335: Rename attributesMap.go
https://github.com/open-telemetry/opentelemetry-go/blob/63a11144cf5a5fd185e40eba7dd22bce2b2980ef/sdk/trace/attributesMap.go#L1

While AFAIK there isn't a standard convention for naming "regular" (i.e. non-test) source files in Go, we seem to be using `foo_bar.go` throughout the codebase and this specific file seems odd.

We might want to rename this file to `attributes_map.go` or `attributesmap.go`.

### PR DESCRIPTION
**Pull request by dstdfx**
This PR adds missing tests for attributesMap and also fixes #1335

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: sdk/trace/attributesmap_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	for i := 128; i < 256; i++ {
+		_, ok := attrMap.attributes[label.Key(fmt.Sprintf("test-key-%d", i))]
+		if !ok {
+			t.Fatal("should not be dropped")
+		}
+	}
+}
+
+func TestAttributesMap_GetOldest_RemoveOldest(t *testing.T) {
```
Comment (MrAlias) [inline sdk/trace/attributesmap_test.go:53]: nit: can the underscores be avoided in the test name?

```suggestion
func TestAttributesMapGetOldestRemoveOldest(t *testing.T) {
```

similar below.

**Review by MrAlias — APPROVED** [review]

File: sdk/trace/attributesmap_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+)
+
+func TestAttributesMap(t *testing.T) {
+	attrMap := newAttributesMap(128)
+
+	for i := 0; i < 256; i++ {
+		attrMap.add(label.Int(fmt.Sprintf("test-key-%d", i), i))
+	}
+	if attrMap.capacity != 128 {
+		t.Fatalf("bad capacity: %v", attrMap.capacity)
```
Comment (dashpole) [inline sdk/trace/attributesmap_test.go:32]: Here and elsewhere, prefer using `Errorf` if the test can continue checking other conditions.

Comment (dashpole) [inline sdk/trace/attributesmap_test.go:32]: Here and elsewhere, try to use the got, want pattern.  E.g. https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/trace/trace_test.go#L707

**Review by XSAM — APPROVED** [review]

File: sdk/trace/attributesmap_test.go [hunk 3]
```diff
@@ -22,30 +22,35 @@ import (
...
 	}
 
-	for i := 0; i < 128; i++ {
-		_, ok := attrMap.attributes[label.Key(fmt.Sprintf("test-key-%d", i))]
+	for i := 0; i < wantCapacity; i++ {
+		key := label.Key(fmt.Sprintf(testKeyFmt, i))
+		_, ok := attrMap.attributes[key]
 		if ok {
-			t.Fatal("should be dropped")
+			t.Errorf("key %q should be dropped", testKeyFmt)
```
Comment (dashpole) [inline sdk/trace/attributesmap_test.go:46]: I think you want to print out `key` here.

**Review by dashpole — APPROVED** [review]

**Pull request merged**